lib/src/protofier.ts

```diff
@@ -144,6 +156,69 @@ export class Protofier {
     }
   }
 
+  /** Converts `calculation` to its protocol buffer representation. */
+  private protofyCalculation(
+    calculation: SassCalculation
+  ): proto.Value_Calculation {
+    return new proto.Value_Calculation({
+      name: calculation.name,
+      arguments: calculation.arguments
+        .map(this.protofyCalculationValue.bind(this))
+        .toArray(),
+    });
+  }
+
+  /** Converts a CalculationValue that appears within a `SassCalculation` to
+   * its protocol buffer representation. */
+  private protofyCalculationValue(
+    value: Object
+  ): proto.Value_Calculation_CalculationValue {
+    const result = new proto.Value_Calculation_CalculationValue();
+    if (value instanceof SassCalculation) {
+      result.value = {
+        case: 'calculation',
+        value: this.protofyCalculation(value),
+      };
+    } else if (value instanceof CalculationOperation) {
+      result.value = {
+        case: 'operation',
+        value: new proto.Value_Calculation_CalculationOperation({
+          operator: this.protofyCalculationOperator(value.operator),
+          left: this.protofyCalculationValue(value.left),
+          right: this.protofyCalculationValue(value.right),
+        }),
+      };
+    } else if (value instanceof CalculationInterpolation) {
+      result.value = {case: 'interpolation', value: value.value};
+    } else if (value instanceof SassString) {
+      result.value = {case: 'string', value: value.text};
+    } else if (value instanceof SassNumber) {
+      // @ts-ignore
+      result.value = this.protofy(value).value;
```

Comment:
I couldn't get TS to see that when `value` is a `SassNumber` I should just be able to protofy it and get a valid return value. The full error is:

```
Type '{ value: Value_String; case: "string"; } | { value: Value_Number; case: "number"; } | { value: Value_RgbColor; case: "rgbColor"; } | { value: Value_HslColor; case: "hslColor"; } | ... 8 more ... | { ...; }' is not assignable to type '{ value: Value_Number; case: "number"; } | { value: string; case: "string"; } | { value: string; case: "interpolation"; } | { value: Value_Calculation_CalculationOperation; case: "operation"; } | { ...; } | { ...; }'.
  Type '{ value: Value_String; case: "string"; }' is not assignable to type '{ value: Value_Number; case: "number"; } | { value: string; case: "string"; } | { value: string; case: "interpolation"; } | { value: Value_Calculation_CalculationOperation; case: "operation"; } | { ...; } | { ...; }'.
    Type '{ value: Value_String; case: "string"; }' is not assignable to type '{ value: string; case: "string"; }'.
      Types of property 'value' are incompatible.
        Type 'Value_String' is not assignable to type 'string'.ts(2322)
```

It seems like there's a mismatch between TS's `string` type and `Value_String`, but it shouldn't be relevant in this case because we have narrowed down the type to `SassNumber`. Any pointers on how to fix it?

Reply:
I think the simplest solution here would just be to factor out a `protofyNumber()` function and call that from both here and `protofy()`.